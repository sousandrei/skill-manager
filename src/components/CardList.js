import React from 'react'
import PropTypes from 'prop-types'

import Card from './Card'

const CardList = ({
	skills
}) =>
	(
		<div className='card-list'>
			{skills.map((s, i) =>
				<Card key={i} index={i} skill={s} />)}
		</div>
	)

CardList.propTypes = {
	skills: PropTypes.array
}

export default CardList

