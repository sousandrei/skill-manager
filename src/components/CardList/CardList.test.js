import React from 'react'
import { shallow } from 'enzyme'

import CardList from './CardList'
import SkillMock from '../../mocks/Skill.mock'

test('should render cardlist', () => {
	const skills = [SkillMock, SkillMock, SkillMock]

	const wrapper = shallow(<CardList skills={skills} />)

	expect(wrapper).toMatchSnapshot()
})

