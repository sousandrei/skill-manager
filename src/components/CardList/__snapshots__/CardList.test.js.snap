// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render cardlist 1`] = `
<div
  className="card-list"
>
  <Connect(Card)
    index={0}
    key="0"
    skill={
      Object {
        "experience": "1 year",
        "id": 123,
        "name": "skill-1",
      }
    }
  />
  <Connect(Card)
    index={1}
    key="1"
    skill={
      Object {
        "experience": "1 year",
        "id": 123,
        "name": "skill-1",
      }
    }
  />
  <Connect(Card)
    index={2}
    key="2"
    skill={
      Object {
        "experience": "1 year",
        "id": 123,
        "name": "skill-1",
      }
    }
  />
</div>
`;
